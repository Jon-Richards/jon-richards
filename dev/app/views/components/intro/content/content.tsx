import * as React from 'react';
const CSS = require('./content.scss');
import { SkillFilters, SkillFilter } from './skill_filters';

type Props = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** Array of skills by which portfolio pieces can be ordered. */
  skills?: SkillFilter[];
};

/** Renders the content for the intro section. */
function IntroContent(props: Props) {
  const { 
    title,
    subtitle,
    skills = []
  } = props;

  return (
    <div className={CSS['content']}>
      <div className={CSS['headline']}>
        <h1 className={CSS['title']}>{title}</h1>
        <h2 className={CSS['subtitle']}>{subtitle}</h2>
      </div>
      <div className={CSS['skills']}>
        <SkillFilters skills={skills} />
      </div>
    </div>
  );
}

const introContentMemo = React.memo<Props>(IntroContent);

export { introContentMemo as IntroContent };