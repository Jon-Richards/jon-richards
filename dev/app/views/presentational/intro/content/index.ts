export { Content, ContentProps } from './content';
