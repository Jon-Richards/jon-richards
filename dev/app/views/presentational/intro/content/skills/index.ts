export { Skills, SkillsProps } from './skills';
