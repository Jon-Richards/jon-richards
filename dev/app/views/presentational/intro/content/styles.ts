import { Breakpoints } from 'Config/styles/breakpoints';
import { Fonts } from 'Config/styles/fonts';
import { Colors } from 'Config/styles/colors';

/** Styling for the Info Content component. */
export const STYLES = {
  content: `
    display: flex;
    flex-direction: column;
    padding: 2rem;

    @media (min-width: ${Breakpoints[720]}) {
      flex-direction: row;
    }
  `,
  headline: `
    display: flex;
    width: auto;
    flex-grow: 0;
    flex-shrink: 0;
    flex-direction: column;
    align-items: flex-start;

    @media (min-width: ${Breakpoints[480]}) {
      width: auto;
      margin: auto;
    }

    @media (min-width: ${Breakpoints[720]}) {
      width: auto;
    }
  `,
  title: `
    display: flex;
    margin-bottom: 0.2rem;
    font-family: ${Fonts.HEEBO.fontFamily};
    font-size: 2.8rem;
    color: ${Colors.SWATCH_2.value};
    line-height: 3.2rem;
  `,
  subtitle: `
    display: flex;
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.regular};
    font-size: 1.4rem;
    color: ${Colors.SWATCH_2.value};
  `,
  children: `
    display: none;

    @media (min-width: ${Breakpoints[480]}) {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      width: auto;
      padding: 1rem 0rem 0rem 0rem;
      margin: 1.4rem 0rem 0rem 0rem;
      border-top-width: thin;
      border-color: ${Colors.SWATCH_2.withLightness(40).value};
      border-style: dotted;
    }

    @media (min-width: ${Breakpoints[720]}) {
      padding: 0rem 0rem 0rem 1rem;
      margin: 0rem 0rem 0rem 1.4rem;
      border-top-width: 0rem;
      border-left-width: thin;
    }
  `
};