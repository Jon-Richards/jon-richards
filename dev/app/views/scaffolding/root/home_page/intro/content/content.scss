@import '~scss-standards';

.content {
  display: flex;
  flex-direction: column;
  padding: 2rem;

  @media (min-width: std-get-breakpoint(720)) {
    flex-direction: row;
  }
}

.headline {
  display: flex;
  width: auto;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: column;
  align-items: flex-start;

  @media (min-width: std-get-breakpoint(480)) {
    width: auto;
    margin: auto;
  }

  @media (min-width: std-get-breakpoint(720)) {
    width: auto;
  }
}

.title {
  display: flex;
  margin-bottom: 0.2rem;
  @include std-font-1;
  font-size: 2.8rem;
  color: std-get-swatch(2);
  line-height: 3.2rem;
}

.subtitle {
  display: flex;
  @include std-font-2;
  font-size: 1.4rem;
  color: std-get-swatch(2);
}

.skills {
  display: none;

  @media (min-width: std-get-breakpoint(480)) {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: auto;
    padding: 1rem 0rem 0rem 0rem;
    margin: 1.4rem 0rem 0rem 0rem;
    border-top-width: thin;
    border-color: std-get-swatch(2, -60);
    border-style: dotted;
  }

  @media (min-width: std-get-breakpoint(720)) {
    padding: 0rem 0rem 0rem 1rem;
    margin: 0rem 0rem 0rem 1.4rem;
    border-top-width: 0rem;
    border-left-width: thin;
  }
}