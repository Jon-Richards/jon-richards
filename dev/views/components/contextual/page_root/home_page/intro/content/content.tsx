import * as React from 'react';
const CSS = require('./content.scss');
import { SkillFilters, SkillFilter } from './skill_filters';

type Props = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** Array of skills by which portfolio pieces can be ordered. */
  skills?: SkillFilter[];
};

/** Renders the content for the intro section. */
export const CONTENT = React.memo<Props>(props => {
  const { 
    title,
    subtitle,
    skills = []
  } = props;

  return (
    <div className={CSS['content']}>
      <div className={CSS['headline']}>
        <h1 className={CSS['title']}>{title}</h1>
        <h2 className={CSS['subtitle']}>{subtitle}</h2>
      </div>
      <div className={CSS['skills']}>
        <SkillFilters skills={skills} />
      </div>
    </div>
  );
});