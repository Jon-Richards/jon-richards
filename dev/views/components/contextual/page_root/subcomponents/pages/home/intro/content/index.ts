export { CONTENT as Content } from './content';