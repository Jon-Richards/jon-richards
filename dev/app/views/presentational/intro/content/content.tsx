/** @jsx jsx */

import * as React from 'react';
import { css, jsx } from '@emotion/core';
import { STYLES } from './styles';
import { Skills, SkillsProps } from './skills';

type ContentProps = {
  /** The intro title. */
  title: React.ReactFragment;
  /** The subtitle. */
  subtitle: React.ReactFragment;
  /** A list of skills. */
  skills: SkillsProps['skills'];
};

/** Renders the content for the intro section. */
function Content (props: ContentProps) {
  const {
    title,
    subtitle,
    skills,
  } = props;

  return (
    <div css={css(STYLES.content)}>
      <div css={css(STYLES.headline)}>
        <h1 css={css(STYLES.title)}>{title}</h1>
        <h2 css={css(STYLES.subtitle)}>{subtitle}</h2>
      </div>
      {skills.length > 0 && (
        <div css={css(STYLES.children)} data-testid="intro_content_skills">
          <Skills skills={skills} />
        </div>
      )}
    </div>
  );
}

const contentMemo = React.memo<ContentProps>(Content);

export { contentMemo as Content, ContentProps };
