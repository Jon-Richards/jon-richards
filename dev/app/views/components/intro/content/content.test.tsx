import * as React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Content } from './content';

describe('The Intro\'s Content component', () => {
  it('Should match the snaptshot.', () => {
    const test_1 = shallow(
      <Content title="Foo" subtitle="Bar" skills={[]} />
    );

    expect(test_1).toMatchSnapshot();
  });

  it('Should render according to its props.', () => {
    const test_1 = shallow(
      <Content title="Foo" subtitle="Bar" skills={[]} />
    );

    expect(test_1.find('.title').contains('Foo')).toBe(true);
    expect(test_1.find('.subtitle').contains('Bar')).toBe(true);
  });
});