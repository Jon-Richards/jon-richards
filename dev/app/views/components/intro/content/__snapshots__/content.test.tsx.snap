// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Intro's Content component Should match the snaptshot. 1`] = `
<div
  className="content"
>
  <div
    className="headline"
  >
    <h1
      className="title"
    >
      Foo
    </h1>
    <h2
      className="subtitle"
    >
      Bar
    </h2>
  </div>
</div>
`;
